frontend/src/data/parseReportData.ts

```diff
@@ -1,3 +1,15 @@
+export const parseReportWithDiff = (
```

Comment:
It will require a small bit of additional refactoring, but you shouldn't need to download the whole previous report to do this -- all you need is the first line of it! So you should be able to just retrieve the first line to get only the implentations from yesterday. On the backend this corresponds to the `RunMetadata` class, so we may need a version of that in the UI.

Reply:
@Julian But in the case of frontend, the report would be present remotely and I am not sure if we can just fetch the first line for this file, the whole file would have to be downloaded first then we can discard the contents except the first line, no ?

Reply:
No certainly not :) -- HTTP is built precisely for these kinds of things. It supports even streaming, so you open requests to somewhere that may produce infinite data and just read some of it. Or if it's helpful conceptually, the mental model you probably should have is that when you transfer some data from one place to the other that's like someone who has a copy of it reading you all of its contents over the phone from beginning to end -- you can of course now not get all of them, just hang up the phone after you hear what you want.

Now which Javascript APIs support this that I don't know offhand :), so would require some looking. From a first google, it seems possible that the API might be the same one we're using (`fetch()`) but instead of awaiting it which will wait for the whole body, maybe [`response.body`](https://developer.mozilla.org/en-US/docs/Web/API/Request/body) which returns a stream (that then you can read from until you see a newline). There may though be a middle-level API in between the two which gives you something that already does a bit more for you, but I'm not sure -- I'd try to use `response.body` and see if it looks like you end up having to write more than 3-4 lines of code.

Reply:
I was able to implement a stream reader which can read the stream until it encounters a newline in the text.